CONTRIBUTING.md

```diff
@@ -0,0 +1,193 @@
+# Contributing to Numberscope
+
+## If you are new to software development, watch the video tutorials
+
+Watch the video series on contributing to Numberscope
+[here](https://www.youtube.com/playlist?list=PLA4KIQBQQRb5ccOdr9v0iLw_fKHup1PkU).
+This video series will introduce you to Numberscope, GitHub, the software we
+use in the Numberscope project, and the workflow for contributing to
+Numberscope.
+
+## If you have software dev experience, here's the TL;DR
+
+1. Create a fork of the Numberscope repository you'd like to contribute to.
+2. Create a branch within that repository.
+3. Do your work on the branch you created. (Make sure your pre-commit hooks
+   are working.)
+4. When you think your code is in a good state, submit a pull request.
+
+## Scenarios and what to do about them
```

Comment:
Maybe after the TL;DR but before the scenarios, you should put a link to a really good git overview on the right level of detail for us, so that people who want the "big git picture" can look at that.  It's something that I'm assuming is done well in various places, so we probably don't need to repeat that general rundown ourselves.